api/worker.go

```diff
@@ -66,11 +67,9 @@ type RHPRenewResponse struct {
 
 // RHPFundRequest is the request type for the /rhp/fund endpoint.
 type RHPFundRequest struct {
-	Contract types.FileContractRevision `json:"contract"`
-	HostKey  consensus.PublicKey        `json:"hostKey"`
-	HostIP   string                     `json:"hostIP"`
-	Account  rhpv3.Account              `json:"account"`
-	Amount   types.Currency             `json:"amount"`
+	ContractID types.FileContractID `json:"contractID"`
+	HostKey    consensus.PublicKey  `json:"hostKey"`
+	Amount     types.Currency       `json:"amount"`
```

Comment:
Removing the HostIP and revision from the request since it's just more consistent with the other endpoints and a lot easier to use.
The worker can fetch the required data from the bus and eventually even cache it. The caller only needs to worry what host to fund an EA with, how much money to put in and what contract to use

Reply:
If you want we can keep the stateless ones under `/rhp` (since that sounds very low-level anyway) and keep higher level, stateful variations under `/accounts` such as `/accounts/fund`.